2001-01-16  Jakub Jelinek  <jakub@redhat.com>

	* config/i386/i386.c (fcmov_comparison_operator): Only initialize
	inmode after checking GET_CODE (op).

--- gcc/config/i386/i386.c.jj	Thu Jan  4 18:08:12 2001
+++ gcc/config/i386/i386.c	Tue Jan 16 17:57:48 2001
@@ -1322,7 +1322,6 @@ fcmov_comparison_operator (op, mode)
     register rtx op;
     enum machine_mode mode;
 {
-  enum machine_mode inmode = GET_MODE (XEXP (op, 0));
   if (mode != VOIDmode && GET_MODE (op) != mode)
     return 0;
   switch (GET_CODE (op))
@@ -1330,7 +1329,7 @@ fcmov_comparison_operator (op, mode)
     case EQ: case NE:
       return 1;
     case LTU: case GTU: case LEU: case ORDERED: case UNORDERED: case GEU:
-      if (inmode == CCmode)
+      if (GET_MODE (XEXP (op, 0)) == CCmode)
 	return 1;
       return 0;
     default:
